java/ql/lib/semmle/code/java/dataflow/internal/FlowSummaryImpl.qll

```diff
@@ -296,11 +296,21 @@ module Public {
     predicate hasProvenance(Provenance provenance) { provenance = "manual" }
   }
 
-  /** A callable where there is no flow via the callable. */
-  class NeutralCallable extends SummarizedCallableBase {
+  /**
+   * A callable where there is no flow via the callable.
+   */
+  class NeutralSummaryCallable extends NeutralCallable {
+    NeutralSummaryCallable() { this.getKind() = "summary" }
+  }
+
+  /**
+   * A callable that has a neutral model.
+   */
+  class NeutralCallable extends ModelingCandidateCallable {
```

Comment:
IMO it is the naming of the file `FlowSummaryImpl.qll` (and friends) that seem incorrect as the file also includes predicates (and probably has done so for a long time) related to sources, sinks (and now also neutrals). As such, I don't think we need to split the file because (as an example) the `External` module in the file contains predicates which uses several of modelling predicates in conjunction with each other (as `relevantSpec`).
If you have uses cases for `SourceCallable` and friends, this is probably still the right file, but we can consider renaming the files and get rid of the `FlowSummary` part as it doesn't properly describes the content.